[vcpkg] [cudnn] [msmpi] [openmpi] Update VMSS

BillyONeal opened the pull request:
This update is primarily to take fixes pushed by the Azure DevOps team for the Linux agents. Additionally, parts of the following VM update requests have been rolled up into this PR:

* https://github.com/microsoft/vcpkg/pull/11321
* https://github.com/microsoft/vcpkg/pull/11378
* https://github.com/microsoft/vcpkg/pull/11373
* https://github.com/microsoft/vcpkg/pull/11596

Special thanks to @JackBoosY for fixing a bunch of ports detected by this change.

Resolves #11451

cenit committed:
[msmpi] update to 10.1

Previously https://github.com/microsoft/vcpkg/pull/11321

cenit committed:
[openmpi] update to 4.0.3

cenit committed:
[openmpi] remove unnecessary build artifacts

BillyONeal commented on scripts/azure-pipelines/windows/provision-image.ps1:
```diff
@@ -112,14 +112,14 @@ $Workloads = @(
-  'occupancy_calculator_10.1 fortran_examples_10.1'
+$MpiUrl = 'https://download.microsoft.com/download/a/5/2/a5207ca5-1203-491a-8fb8-906fd68ae623/msmpisetup.exe'
+
+$CudaUrl = 'http://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda_10.2.89_441.22_win10.exe'
```
Hmmm this should have been left as HTTPS

cenit replied:
You are right sorry. I copy/pasted from browser without noticing

MVoz committed:
[networkdirect] Add the Windows WDK to VM

Part of https://github.com/microsoft/vcpkg/pull/11373

BillyONeal committed:
[llvm] Add LLVM installed by Visual Studio for potential consumption in a future triplet.

Also unblocks https://github.com/microsoft/vcpkg/pull/11655 by getting clang-format on the boxes.

BillyONeal committed:
Use a VM series with a 400GB temporary disk which replaces the 'installed' disk.

BillyONeal committed:
Log provision results for inspection later.

BillyONeal committed:
Fix incorrect Write-Progress in create-vmss.ps1.

BillyONeal committed:
Remove Linux VM attempt to put the work tree on /mnt.

BillyONeal committed:
Remove no longer needed /home manipulation in provisioning script.

BillyONeal commented:
Depends on https://github.com/microsoft/vcpkg/pull/11494

BillyONeal commented:
Depends on https://github.com/microsoft/vcpkg/pull/11570

BillyONeal commented:
Depends on https://github.com/microsoft/vcpkg/pull/11571

JackBoosY commented:
That's hard work. Thanks @BillyONeal !

Voskrese commented:
SDK new ver msvc new sdk ?

![2020-05-27_155909](https://user-images.githubusercontent.com/10104740/82993607-71b9ce00-a033-11ea-9af2-fc0cecb0f409.png)

Microsoft.VisualStudio.Component. **Windows10SDK.19041**

https://github.com/jpark37/virtual-environments/commit/8b4b249baa55ab2142d3e9ca89383262cccca961

BillyONeal committed:
[qt5] Add additional prerequisites for qt components.

Part of https://github.com/microsoft/vcpkg/pull/11596/

BillyONeal committed:
Skip Add-MPPreference when the AV is not installed.

BillyONeal committed:
[cuda] Remove clearly unused link to nonexistent cuda target.

BillyONeal committed:
Combine some apt installs, remove /etc/debian_version workaround, defend against no StorageAccountName et al. set.

JackBoosY commented:
/azp run

BillyONeal committed:
Update pools

JackBoosY commented:
/azp run

BillyONeal commented:
/azp run

Voskrese commented:
```
-- Downloading https://sourceforge.net/projects/aifftools/files/libaiff/LibAiff%205.0/libaiff-5.0-release.tar.gz... Failed. Status: 7;"Couldn't connect to server"
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:175 (message):
      
      Failed to download file.
      If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
      variables to "https://user:password@your-proxy-ip-address:port/".
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues
```

```
-- Downloading https://sourceforge.net/projects/libusb-win32/files/libusb-win32-releases/1.2.6.0/libusb-win32-src-1.2.6.0.zip/download... Failed. Status: 7;"Couldn't connect to server"
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:175 (message):
      
      Failed to download file.
      If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
      variables to "https://user:password@your-proxy-ip-address:port/".
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues
```

re-run ???

BillyONeal committed:
[boost-context] [boost-coroutine] [ppconsul] [nettle] Update baselines.

BillyONeal committed:
Merge delete tombstones and PR build into the same pipeline, controlled by a variable.

BillyONeal committed:
[sdl2] Apply @JackBoosY 's fix for SDL2 on Visual Studio 2019 16.6.

BillyONeal committed:
[qt5] Baseline qt5-translations for now as Qt was broken by Visual Studio 2019 16.6

strega-nil commented on scripts/azure-pipelines/azure-pipelines.yml:
```diff
@@ -2,52 +2,88 @@
-      poolName: $(linux-pool)
+  windows-pool: 'PrWin-2020-06-04'
+  linux-pool: 'PrLin-2020-06-08'
+  delete-tombstones-first: false
```
How does one set this variable?

BillyONeal replied:
You can make a PR that turns it on then turns it back off, or you can set it with the web interface.
